src/test/kotlin/com/atlassian/performance/tools/report/api/judge/RelativeNonparametricPerformanceJudgeTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.atlassian.performance.tools.report.api.judge
+
+import com.atlassian.performance.tools.jiraactions.api.EDIT_ISSUE
+import com.atlassian.performance.tools.report.api.ShiftedDistributionRegressionTest
+import com.atlassian.performance.tools.report.api.result.DurationData
+import com.atlassian.performance.tools.report.api.result.EdibleResult
+import com.atlassian.performance.tools.report.junit.FailedActionJunitReport
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when` as whenever
+
+class RelativeNonparametricPerformanceJudgeTest {
+    private fun <T> any(type: Class<T>): T = Mockito.any<T>(type)
+
+    @Test
+    fun shouldProduceExpectedFailedActionReportMessage() {
+        // given
+        val actionToleranceRatio = 0.1f
+        val shiftedDistributionRegressionTest = mock(ShiftedDistributionRegressionTest::class.java).apply {
+            whenever(isExperimentRegressed(any(Double::class.java))).thenReturn(true)
+        }
+        val shiftedDistributionRegressionTestProvider = mock(ShiftedDistributionRegressionTestProvider::class.java).apply {
+            whenever(
+                get(
+                    baseline = any(DoubleArray::class.java),
+                    experiment = any(DoubleArray::class.java),
+                    mwAlpha = any(Double::class.java),
+                    ksAlpha = any(Double::class.java)
+                )
+            ).thenReturn(shiftedDistributionRegressionTest)
+        }
+        val tested = RelativeNonparametricPerformanceJudge.Builder()
+            .significance(0.05)
+            .actionMetricsReader {
+                mapOf(EDIT_ISSUE.label to mock(DurationData::class.java).apply {
+                    whenever(statsValues()).thenReturn(doubleArrayOf())
+                })
+            }
+            .shiftedDistributionRegressionTestProvider(shiftedDistributionRegressionTestProvider)
+            .build()
+        // when
+        val result = tested.judge(
+            toleranceRatios = mapOf(EDIT_ISSUE to actionToleranceRatio),
+            baselineResult = mock(EdibleResult::class.java),
+            experimentResult = mock(EdibleResult::class.java)
+        )
+        // then
+        assertThat(result.reports).hasSize(1)
+        assertThat((result.reports[0] as FailedActionJunitReport).toXml(testClassName = "does not matter"))
+            .contains("There is a regression in [Full Edit Issue] with 95% confidence level. Regression is larger than allowed +10.00% tolerance")
```

Comment:
The test was added in the same commit as the change in `src/main`, so it's unclear what the improvement is.

Reply:
I'm not the PR author here, but I personally don't quite understand this comment. I'd appreciate if you could elaborate, so I could learn from this contribution as well.

Reply:
I assume that the PR is improving the message, which is tested here. But this test didn't exist before the improvement. So we only see the new version. If the test and fix commits were separate, then the diff would show how the message was improved.

Reply:
@dagguh would you see it valid to merge such split commit structure as well or would that operation of sliting a test be only to demonstrate the fix for the review purpose? I'm trying to figure out the algorythm here.

Reply:
It doesn't have to be done via commits. I'm just looking for the value proposition anywhere to learn whether it's worth the price.